samples/nrf9160/modem_shell/src/gnss/gnss_supl_support.c

```diff
@@ -22,98 +22,83 @@
 #define SUPL_SERVER "supl.google.com"
 #define SUPL_SERVER_PORT 7276
 
-/* Number of getaddrinfo attempts */
-#define GAI_ATTEMPT_COUNT 3
-
-static int supl_fd;
+static int supl_fd = -1;
 
 extern const struct shell *shell_global;
 
 int open_supl_socket(void)
 {
 	int err = -1;
-	int proto;
-	int gai_cnt = 0;
-	uint16_t port;
-	struct addrinfo *addr;
 	struct addrinfo *info;
 
-	proto = IPPROTO_TCP;
-	port = htons(SUPL_SERVER_PORT);
-
 	struct addrinfo hints = {
-		.ai_family = AF_INET,
-		.ai_socktype = SOCK_STREAM,
-		.ai_protocol = proto,
-		/* Either a valid,
-		 * NULL-terminated access point name or NULL.
-		 */
-		.ai_canonname = NULL,
+		.ai_family = AF_UNSPEC, /* Both IPv4 and IPv6 addresses accepted. */
+		.ai_socktype = SOCK_STREAM
 	};
 
-	/* Try getaddrinfo many times, sleep a bit between retries */
-	do {
-		err = getaddrinfo(SUPL_SERVER, NULL, &hints, &info);
-		gai_cnt++;
-
-		if (err) {
-			if (gai_cnt < GAI_ATTEMPT_COUNT) {
-				/* Sleep between retries */
-				k_sleep(K_MSEC(1000 * gai_cnt));
-			} else {
-				/* Return if no success after many retries */
-				shell_error(
-					shell_global,
-					"GNSS: Failed to resolve hostname %s on IPv4, errno: %d)",
-					SUPL_SERVER, errno);
-
-				return -1;
-			}
-		}
-	} while (err);
-
-	/* Create socket */
-	supl_fd = socket(AF_INET, SOCK_STREAM, proto);
-	if (supl_fd < 0) {
-		shell_error(shell_global,
-			    "GNSS: Failed to create socket, errno %d", errno);
-		goto cleanup;
-	}
-
-	struct timeval timeout = {
-		.tv_sec = 1,
-		.tv_usec = 0,
-	};
-
-	err = setsockopt(supl_fd, NRF_SOL_SOCKET, NRF_SO_RCVTIMEO, &timeout,
-			 sizeof(timeout));
+	err = getaddrinfo(SUPL_SERVER, NULL, &hints, &info);
 	if (err) {
-		shell_error(shell_global,
-			    "GNSS: Failed to set socket timeout, errno %d",
-			    errno);
-		goto cleanup;
+		shell_error(
+			shell_global,
+			"GNSS: Failed to resolve hostname %s, errno: %d)",
+			SUPL_SERVER, errno);
+
+		return -1;
 	}
 
 	/* Not connected */
 	err = -1;
 
-	for (addr = info; addr != NULL; addr = addr->ai_next) {
+	for (struct addrinfo *addr = info; addr != NULL; addr = addr->ai_next) {
+		char ip[INET6_ADDRSTRLEN] = { 0 };
 		struct sockaddr *const sa = addr->ai_addr;
 
 		switch (sa->sa_family) {
 		case AF_INET6:
-			((struct sockaddr_in6 *)sa)->sin6_port = port;
+			((struct sockaddr_in6 *)sa)->sin6_port = htons(SUPL_SERVER_PORT);
 			break;
 		case AF_INET:
-			((struct sockaddr_in *)sa)->sin_port = port;
+			((struct sockaddr_in *)sa)->sin_port = htons(SUPL_SERVER_PORT);
 			break;
 		}
 
+		supl_fd = socket(sa->sa_family, SOCK_STREAM, IPPROTO_TCP);
+		if (supl_fd < 0) {
+			shell_error(shell_global,
+				    "GNSS: Failed to create socket, errno %d", errno);
+			goto cleanup;
+		}
+
+		struct timeval timeout = {
+			.tv_sec = 1,
```

Comment:
Not demanding change here, and not familiar with SUPL, but 1sec sounds quite  a short timeout, especially with hard network conditions / nbiot. 
Would it worth to have separate config for this?

Reply:
The timeout is 1s because of a peculiarity of the SUPL library, the library expects the provided read function to timeout after 1 second if no data is received. The library does call the read function several times, but still, it's a bit strange. Anyway, changing this would require changing the SUPL library and the library API as well.

Maybe it's a good idea to add a comment here, because I almost changed the code myself before realizing that it needs to be like this. :) I'll add a comment.